### ISSUE TITLE
The function dotprint() is omitted from the sympy.printing module

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#15088]
I've recently been following up the [tutorial](http://docs.sympy.org/latest/tutorial/printing.html) of printing.

And while the tutorial suggests me to import the method in following way
```from sympy.printing.dot import dotprint```
and regard it as
```dotprint()```

I would also like to use the method in hierarchically accessible way such as like in the following
```import sympy```
and access it as something like
```sympy.printing.dot.dotprint()```
```sympy.printing.dotprint()```

However, I cannot access the function directly from `sympy.printing`, if I import sympy in latter way, while the functionality of the method suggests that it should be put on same level as other printing methods which are publically accessible from `sympy.printing`.

For testing my issues out
Open up a new python shell and type out
```
import sympy

dir(sympy.printing)
```
and check if anything related to `dot` or `dotprint` appears under it.

I have checked the `__init__.py` files from the `sympy` and `sympy.printing` package and found out that the `dot.py` module seems completely isolated from the packages.
If there is no specific reason for this convention, and if the issue needs to be fixed, I would also be glad to offer some help

**Comment by asmeurer** [comment]
Several functions in SymPy aren't included in `__init__.py` files, meaning they can only be imported directly. I guess `dotprint` is one of these. However, since it is mentioned in the tutorial, we should probably include it. 

This should be easy to fix. Just add `dotprint` to `sympy/printing/__init__.py`.

**Comment by sylee957** [comment]
This issue has already been merged

**Comment by asmeurer** [comment]
In the future you can add "Fixes #15088" to the pull request description or a commit message and the issue will close automatically when the pull request is merged.

**Comment by asmeurer** [comment]
Actually that PR didn't make `sympy.dotprint` work, because the main `sympy/__init__.py` file does explicit imports from `.printing`.

**Comment by asmeurer** [comment]
Yes, it is fine. I think the reason there were explicit imports in `sympy/__init__.py` is that `import *` would also import the submodule names, which would mask some builtins (`str` and `repr` in particular). But the solution to this is to define `__all__` in `sympy/printing/__init__.py`. 

I have a fix at https://github.com/sympy/sympy/pull/15162
